### PR TITLE
refactor: add listen param and drop podman-in-podman

### PR DESCRIPTION
**Pull request by cmdoret**
**Summary**
Updates the workflow and documentation to reflect the new practices in LUCID components:

* event-based mode (via `watchPath`) using `--listen=true` option
* simpler architecture: no podman-in-podman (1 input = zip file with group of patients)

**Additional changes**
This PR simplifies the workflow architecture by dropping the podman-in-podman setup.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -133,11 +63,38 @@ First clone the repository and move into the folder:
...
 
+* `nextflow run -profile standard main.nf`: Run the workflow using the workflow file in the current directory and publicly available containers defined in `conf/containers.yaml`. This is the default profile, and the `-profile` option can therefore be omitted.
 * `nextflow run -profile prod main.nf`: Run the containerized workflow using the latest commit on the repository remote and containers from the private registry.
-* `nextflow run -profile dev main.nf`: Run the containerized workflow using the workflow file in the current directory and publicly available containers defined in `conf/containers.yaml`.
 
+### Execution mode
+
+By default, the workflow will be executed on each zip file present in the input directory.
+
+When the option `--listen=true` is provided, the workflow manager will instead listen continuously for filesystem events and trigger execution whenever a new zip file appears in the input directory.
```
Comment (supermaxiste) [inline README.md]:
```suggestion
When the option `--listen=true` is provided, the workflow manager will instead listen continuously for filesystem events and trigger execution whenever a new zip file appears in the input directory. In this mode, a log will only be generated when the Nextflow execution is manually interrupted, meaning that the exit code in the log will always be 1.
```

File: README.md [hunk 2]
```diff
@@ -133,11 +63,38 @@ First clone the repository and move into the folder:
...
+
```
Comment (supermaxiste) [inline README.md:86]:
```suggestion

### Inputs

This repository includes a small set of example data in `data/raw/test_patients.zip`. If you'd like to use a larger dataset as input, we included `scripts/download_data.sh` to retrieve a synthetic dataset from [Synthea](https://github.com/synthetichealth/synthea) and put it directly in the input folder `data/raw/`.

```

**Review by supermaxiste — COMMENTED** [review]
Very thorough update @cmdoret 🥳 I added a couple of suggestions directly and I have some suggestions below.

**praise:** I ran the workflow locally and also tested `listen` mode, all works as expected
**praise:** Adding the `podman.registry` option in the `nextflow.config` makes the code more readable

**suggestion:** if we're planning to keep the script `migrate_images.sh`, we should 1) mention it in the README and 2) provide an example YAML file. The current description at the top is not super clear to me.

**thoughts (non-blocking):** in the process `generate_triples` there's a series of commands before running RMLstreamer because of relative paths+symlinks. I thought of different alternatives but there's nothing really "clean". I also considered breaking down this process for better digestability but I don't think it would make a difference. My take-away is that RMLStreamer is not great when it comes to paths: either you hardcode them or you `sed` in-place. If you have any additional thought feel free to share!

**Comment by cmdoret** [comment]
Thanks @supermaxiste
migrate_images.sh now uses the containers.conf as input, so that one does not have to maintain 2 identical lists of containers in different files. This was indeed not convenient. The example usage now points to `conf/containers.conf`. 

The script is now mentioned in the readme.

**Comment by cmdoret** [comment]
Yeah I did not have any clean solution either, RMLStreamer's path handling a bit annoying to work with.
The only solution I see would be to implement parameters and absolute paths in RMLStreamer.

**Review by supermaxiste — APPROVED** [review]
Very well done, looks good to me 😎

**Pull request merged**